lib/jekyll-webp/webpGenerator.rb

```diff
@@ -44,57 +45,55 @@ def generate(site)
 
         # Iterate through every image in each of the image folders and create a webp image
         # if one has not been created already for that image.
-        for imgdir in @config['img_dir']
-          imgdir_source = File.join(site.source, imgdir)
-          Jekyll.logger.info "WebP:","Processing #{imgdir_source}"
-          
-          # handle only jpg, jpeg, png and gif
-          Dir.foreach(imgdir_source) do |imgfile|
-          
-              # Skip empty stuff
-              next if imgfile == '.' or imgfile == '..'
-              file_ext = File.extname(imgfile).downcase
-
-              # If the file is not one of the supported formats, exit early
-              next if !@config['formats'].include? file_ext
-
-              # TODO: Do an exclude check
-              
-              # Create the output file path
-              file_noext = File.basename(imgfile, file_ext)
-              outfile_filename = file_noext+ ".webp"
-              outfile_fullpath_webp = File.join(imgdir_source, outfile_filename)
-
-              # Create the full input file name
-              infile_fullpath = File.join(imgdir_source, imgfile)
-          
-              # Check if the file already has a webp alternative?
-              # If we're force rebuilding all webp files then ignore the check
-              # also check the modified time on the files to ensure that the webp file
-              # is newer than the source file, if not then regenerate
-              next if !@config['regenerate'] && File.file?(outfile_fullpath_webp) &&
-                      File.mtime(outfile_fullpath_webp) > File.mtime(infile_fullpath)      
-              
-              if( File.file?(outfile_fullpath_webp) && 
-                  File.mtime(outfile_fullpath_webp) <= File.mtime(infile_fullpath) )
-                Jekyll.logger.info "WebP:", "Change to source image file #{imgfile} detected, regenerating WebP"
-              end
-
-              # Generate the file
-              WebpExec.run(@config['quality'], infile_fullpath, outfile_fullpath_webp)
-              
-              # Keep the webp file from being cleaned by Jekyll
-              site.static_files << WebpFile.new(site, site.dest, imgdir, outfile_filename)
-              file_count += 1
-              
-          end # dir.foreach
+        for source in @config['img_dir']
```

Comment:
local variable was intentionally coded as `imgdir` to avoid confusing it with the site.source